packages/sentry/index.tsx

```diff
@@ -0,0 +1,16 @@
+import * as Sentry from '@sentry/nextjs';
+
+export const sentryFetch: typeof fetch = async (resource, options) => {
+   const context = {
+      resource,
+      credentials: options?.credentials,
+      headers: options?.headers,
+      method: options?.method,
+      parameters: options?.body,
+   };
+   Sentry.configureScope((scope) => scope.setContext('request', context));
```

Comment:
I think this is not quite right. `configureScope()` sets the scope *globally* for all future sentry reports. So every request will be flipping this "request" context around when its placed. The *failure* may happen *after* another request has been initiated and thus will be reported with the wrong context. Given that we are catching the errors here, I vote for at least only adding the scopes in the `catch()` to reduce the chance of reporting with the wrong context.

Reply:
Ah you're right, I think we'd rather use [`withScope`](https://docs.sentry.io/platforms/javascript/guides/nextjs/enriching-events/context/#clearing-context) here so that we drop the `request` context after the `fetch()`.

Reply:
`withScope()` is only for synchronous actions, so I don't think it applies for the promise-returning `fetch()`